xbmc/addons/AddonBuilder.cpp

```diff
@@ -199,13 +197,12 @@ AddonPtr CAddonBuilder::FromProps(CAddonInfo addonInfo)
       return AddonPtr(new CScraper(std::move(addonInfo)));
     case ADDON_SKIN:
       return AddonPtr(new CSkinInfo(std::move(addonInfo)));
+    case ADDON_ADSPDLL:
     case ADDON_VIZ:
     case ADDON_SCREENSAVER:
       return AddonPtr(new CAddonDll(std::move(addonInfo)));
```

Comment:
Sorry for the dump question, do all add-on extenstion points have to use CAddonDll as a base class? If so this makes totally sense to me. First time that I really look into your work.